users/tests/test_user_login.py

```diff
@@ -22,3 +33,20 @@ def test_user_client_should_not_log_in_when_email_is_none(self):
 
     def test_user_client_should_not_log_in_when_password_is_none(self):
         self.assertFalse(self.client.login(email="testuser@codepoets.it", password=None))
+
+    def test_user_should_redirect_to_change_password_form_if_need_change_password_is_true(self):
+        data = {"username": self.user.email, "password": self.password}
+        response = self.client.post(reverse("login"), data=data)
+        self.assertRedirects(response, reverse("password_change_on_login"), status_code=302, target_status_code=200)
+
+    def test_user_should_redirect_to_home_if_need_change_password_is_false(self):
+        data = {"username": self.user.email, "password": self.password}
+        self.user.need_change_password = False
+        self.user.save()
+        response = self.client.post(reverse("login"), data=data)
+        self.assertRedirects(response, reverse("home"), status_code=302, target_status_code=200)
```

Comment:
When you have longer tests it's good practice to divide them for logical parts. You can divide them by using blank line between there parts. For example in this case first 3 lines are your test data setup, then you have request which is the main part of this test and at the end your asserting that response is correct. So in this case you should add blank spaces between 45th and 46th line and between 46th and 47th.

Reply:
Done, however not sure if still necessary after resetting need_change_password in private method.

Reply:
After next changes I've reconsidered and decided to not add spaces since there are only 3 lines now:

    def test_user_should_redirect_to_change_password_form_if_need_change_password_is_true(self):
        self._set_need_change_password()
        response = self.client.post(reverse("login"), data=self.data)
        self.assertRedirects(response, reverse("login_password_change"), status_code=302, target_status_code=200)